application_sandbox/format_feature_flags2/main.cpp

```diff
@@ -0,0 +1,615 @@
+        VK_FORMAT_G8_B8_R8_3PLANE_444_UNORM,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_R10X6_UNORM_PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_R10X6G10X6_UNORM_2PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_R10X6G10X6B10X6A10X6_UNORM_4PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G10X6B10X6G10X6R10X6_422_UNORM_4PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_B10X6G10X6R10X6G10X6_422_UNORM_4PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G10X6_B10X6_R10X6_3PLANE_420_UNORM_3PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G10X6_B10X6R10X6_2PLANE_420_UNORM_3PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G10X6_B10X6_R10X6_3PLANE_422_UNORM_3PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G10X6_B10X6R10X6_2PLANE_422_UNORM_3PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G10X6_B10X6_R10X6_3PLANE_444_UNORM_3PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_R12X4_UNORM_PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_R12X4G12X4_UNORM_2PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_R12X4G12X4B12X4A12X4_UNORM_4PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G12X4B12X4G12X4R12X4_422_UNORM_4PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_B12X4G12X4R12X4G12X4_422_UNORM_4PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G12X4_B12X4_R12X4_3PLANE_420_UNORM_3PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G12X4_B12X4R12X4_2PLANE_420_UNORM_3PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G12X4_B12X4_R12X4_3PLANE_422_UNORM_3PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G12X4_B12X4R12X4_2PLANE_422_UNORM_3PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G12X4_B12X4_R12X4_3PLANE_444_UNORM_3PACK16,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G16B16G16R16_422_UNORM,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_B16G16R16G16_422_UNORM,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G16_B16_R16_3PLANE_420_UNORM,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G16_B16R16_2PLANE_420_UNORM,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G16_B16_R16_3PLANE_422_UNORM,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G16_B16R16_2PLANE_422_UNORM,
+        // Provided by VK_VERSION_1_1
+        VK_FORMAT_G16_B16_R16_3PLANE_444_UNORM,
+    };
+
+    bool success = true;
+    for (auto format : formats) {
+      app()->instance()->vkGetPhysicalDeviceFormatProperties2(
+          app()->device().physical_device(), format, &format_properties2);
+      if ((format_properties2.formatProperties.linearTilingFeatures &
+           format_properties3.linearTilingFeatures) !=
+          format_properties2.formatProperties.linearTilingFeatures) {
```

Comment:
This checks that the set bits in `format_properties2.formatProperties.linearTilingFeatures` is a subset of the set bits in `format_properties3.linearTilingFeatures`. But I don't think that's exactly what we want (e.g. that is satisfied if `format_properties2.formatProperties.linearTilingFeatures==0` and `format_properties3.linearTilingFeatures==UINT64_MAX`).

Instead, does it make sense to check
```
if((format_properties3.linearTilingFeatures & kAllVkFormatFeatureFlagBits) !=
          format_properties2.formatProperties.linearTilingFeatures)
```
where `kAllVkFormatFeatureFlagBits` is the `|` of all the valid `VkFormatFeatureFlagBits` values. At a glance, I think you could define `kAllVkFormatFeatureFlagBits=(1u<<31)-1` (all except the highest bit).

Reply:
You are right. Fixed and thanks!

Reply:
Could I get a review of my last change?